cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/batch/joiner/Join.java

```diff
@@ -0,0 +1,258 @@
+    private final String requiredInputs;
+
+
+    public Config() {
+      this.joinKeys = "joinKeys";
+      this.joinType = "joinType";
+      this.numOfInputs = "numOfInputs";
+      this.fieldsToSelect = "fieldsToSelect";
+      this.fieldsToRename = "fieldsToRename";
+      this.requiredInputs = "requiredInputs";
+    }
+
+    private void validateConfig() {
+      validateNumOfInputs();
+
+      if (joinKeys == null || joinKeys.isEmpty()) {
+        throw new IllegalArgumentException(String.format(
+          "join keys can not be empty or null for plugin %s", PLUGIN_CLASS));
+      }
+
+      Iterable<String> multipleJoinKeys = Splitter.on(':').trimResults().omitEmptyStrings().split(joinKeys);
+      if (Iterables.size(multipleJoinKeys) == 0) {
+        throw new IllegalArgumentException(String.format("There should be atleast %s join keys", numOfInputs));
+      }
+
+      for (String key : multipleJoinKeys) {
+        int numOfStages = 0;
+        Iterable<String> perStageJoinKeys = Splitter.on(',').trimResults().omitEmptyStrings().split(key);
+        if (Iterables.size(perStageJoinKeys) != getNumOfInputs()) {
+          throw new IllegalArgumentException(String.format("Number of join keys should be equal for all the stages " +
+                                                             "for plugin %s", PLUGIN_TYPE));
+        }
+
+        for (String perStageKey : perStageJoinKeys) {
+          Iterable<String> stageKey = Splitter.on('.').trimResults().omitEmptyStrings().split(perStageKey);
+          if (Iterables.size(stageKey) != 2) {
+            throw new IllegalArgumentException(String.format("Join key should be specified in stageName:joinKey format"
+                                                               + " for %s of %s", perStageKey, PLUGIN_TYPE));
+          }
+          numOfStages++;
+        }
+        if (numOfStages != getNumOfInputs()) {
+          throw new IllegalArgumentException(String.format("There should be %s number of keys but found %s",
+                                                           getNumOfInputs(), numOfStages));
+        }
+      }
+    }
+
+    private void validateNumOfInputs() {
+      try {
+        if (Integer.parseInt(numOfInputs) <= 0) {
+          throw new IllegalArgumentException(String.format("Number of inputs to %s should be a positive integer but " +
+                                                             "found %s ", PLUGIN_TYPE, numOfInputs));
+        }
+      } catch (NumberFormatException exception) {
+        throw new IllegalArgumentException(String.format("Number of inputs to %s should be a positive integer but " +
+                                                           "found %s ", PLUGIN_TYPE, numOfInputs));
+      }
+    }
+
+    private Map<String, List<String>> getJoinKeys() {
+      Map<String, List<String>> stageToKey = new HashMap<>();
+      Iterable<String> multipleJoinKeys = Splitter.on(':').trimResults().omitEmptyStrings().split(joinKeys);
```

Comment:
This piece of code is difficult to read, it would be good to add comments to describe what it is doing.